src/charm.py

```diff
@@ -124,6 +137,73 @@ def __init__(self, *args):
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
+    def _is_unit_in_non_active_status(self) -> Optional[StatusBase]:  # noqa: C901
+        """Evaluate and return the unit's current status, or None if it should be active.
+
+        Returns:
+            StatusBase: MaintenanceStatus/BlockedStatus/WaitingStatus
+            None: If none of the conditionals match
+
+        """
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to preform if we're removing the
+            # charm.
+            return BlockedStatus("Scaling is not implemented for this charm")
+
+        if not self._amf_container.can_connect():
+            return MaintenanceStatus("Waiting for service to start")
+
+        if invalid_configs := self._get_invalid_configs():
+            return BlockedStatus(f"The following configurations are not valid: {invalid_configs}")
+
+        if not self.model.get_relation("fiveg-nrf") or self._nrf_relation_breaking:
+            return BlockedStatus("Waiting for fiveg-nrf relation")
+
+        if not self.model.get_relation("database") or self._database_relation_breaking:
+            return BlockedStatus("Waiting for database relation")
```

Comment:
In fact self.model.get_relation("database") is generally returning True because of a  [known issue] (https://bugs.launchpad.net/juju/+bug/1979811).  Although the integration is removed, it still appears in the model. As self._database_relation_breaking return True, so the status is set to Blocked.
I also deployed the charm including this patch and tested it manually.

```
$ juju status --relations
Model                  Controller          Cloud/Region        Version  SLA          Timestamp
test-integration-twnt  microk8s-localhost  microk8s/localhost  3.1.6    unsupported  21:49:59+03:00

App                       Version  Status  Scale  Charm                     Channel  Rev  Address         Exposed  Message
mongodb-k8s                        active      1  mongodb-k8s               5/edge    36  10.152.183.198  no       Primary
sdcore-amf                         active      1  sdcore-amf                           0  10.152.183.22   no       
sdcore-nrf                         active      1  sdcore-nrf                edge      58  10.152.183.58   no       
self-signed-certificates           active      1  self-signed-certificates  beta      33  10.152.183.202  no       

Unit                         Workload  Agent  Address      Ports  Message
mongodb-k8s/0*               active    idle   10.1.146.19         Primary
sdcore-amf/0*                active    idle   10.1.146.7          
sdcore-nrf/0*                active    idle   10.1.146.51         
self-signed-certificates/0*  active    idle   10.1.146.15         

Integration provider                   Requirer                    Interface         Type     Message
mongodb-k8s:database                   sdcore-amf:database         mongodb_client    regular  
mongodb-k8s:database                   sdcore-nrf:database         mongodb_client    regular  
mongodb-k8s:database-peers             mongodb-k8s:database-peers  mongodb-peers     peer     
sdcore-nrf:fiveg-nrf                   sdcore-amf:fiveg-nrf        fiveg_nrf         regular  
self-signed-certificates:certificates  sdcore-amf:certificates     tls-certificates  regular  
self-signed-certificates:certificates  sdcore-nrf:certificates     tls-certificates  regular  

$ juju remove-relation mongodb-k8s:database  sdcore-amf:database 
$ juju status --relations
Model                  Controller          Cloud/Region        Version  SLA          Timestamp
test-integration-twnt  microk8s-localhost  microk8s/localhost  3.1.6    unsupported  21:50:38+03:00

App                       Version  Status   Scale  Charm                     Channel  Rev  Address         Exposed  Message
mongodb-k8s                        active       1  mongodb-k8s               5/edge    36  10.152.183.198  no       Primary
sdcore-amf                         waiting      1  sdcore-amf                           0  10.152.183.22   no       installing agent
sdcore-nrf                         active       1  sdcore-nrf                edge      58  10.152.183.58   no       
self-signed-certificates           active       1  self-signed-certificates  beta      33  10.152.183.202  no       

Unit                         Workload  Agent  Address      Ports  Message
mongodb-k8s/0*               active    idle   10.1.146.19         Primary
sdcore-amf/0*                blocked   idle   10.1.146.7          Waiting for database relation
sdcore-nrf/0*                active    idle   10.1.146.51         
self-signed-certificates/0*  active    idle   10.1.146.15         

Integration provider                   Requirer                    Interface         Type     Message
mongodb-k8s:database                   sdcore-nrf:database         mongodb_client    regular  
mongodb-k8s:database-peers             mongodb-k8s:database-peers  mongodb-peers     peer     
sdcore-nrf:fiveg-nrf                   sdcore-amf:fiveg-nrf        fiveg_nrf         regular  
self-signed-certificates:certificates  sdcore-amf:certificates     tls-certificates  regular  
self-signed-certificates:certificates  sdcore-nrf:certificates     tls-certificates  regular 

```
Juju debug-log:
```
core-amf-0: 21:49:50 INFO unit.sdcore-amf/0.juju-log HTTP Request: PATCH https://10.152.183.1/api/v1/namespaces/test-integration-twnt/services/sdcore-amf "HTTP/1.1 200 OK"
unit-sdcore-amf-0: 21:49:50 INFO unit.sdcore-amf/0.juju-log Kubernetes service 'sdcore-amf' patched successfully
unit-sdcore-amf-0: 21:49:51 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
unit-mongodb-k8s-0: 21:50:36 INFO juju.worker.uniter.operation ran "database-relation-departed" hook (via hook dispatching script: dispatch)
unit-sdcore-amf-0: 21:50:37 INFO juju.worker.uniter.operation ran "database-relation-departed" hook (via hook dispatching script: dispatch)
unit-mongodb-k8s-0: 21:50:37 INFO unit.mongodb-k8s/0.juju-log database:4: Remove relation user: relation-4
unit-mongodb-k8s-0: 21:50:37 INFO unit.mongodb-k8s/0.juju-log database:4: Update relation user: relation-1 on free5gc
unit-mongodb-k8s-0: 21:50:37 INFO unit.mongodb-k8s/0.juju-log database:4: Updating relation data according to diff
unit-sdcore-amf-0: 21:50:37 INFO juju.worker.uniter.operation ran "database-relation-broken" hook (via hook dispatching script: dispatch)
unit-mongodb-k8s-0: 21:50:37 INFO juju.worker.uniter.operation ran "database-relation-broken" hook (via hook dispatching script: dispatch)

```